### DISCUSSION TITLE
Programmatically triggering only works on terminal

### DISCUSSION THREAD

**Discussion by JDuffe** [rxhanson/RectanglePro-Community#418] [Q&A] [answered]
I've searched for a solution to my problem for a while now but couldn't figure it out:

I want to trigger an action from Rectangle Pro using an URL, like `open -g "rectangle-pro://execute-action?name=left-half"`. But when I run that URL from the terminal, the terminal gets resized (which also makes sense to me). 
But how can I use those URLs to control any program other than the terminal? 

(I would guess that the URLs are intended to be used inside a script (shell, bash etc.), rather than to be entered manually in the termial. Would that make any difference though?)
What am I doing wrong here? Appreciate any help!

**Comment by rxhanson** [comment, marked as answer]
Typically these are executed using a different application, like Keyboard Maestro, or anything else that can trigger a script or open a URL. 

If you were running from the Terminal, you would need some way to tell Rectangle Pro which window you want to operate on. Currently the only way that you could achieve something like this is to make App Layouts and trigger those via url.
